### PR TITLE
docs: Update existing reference navigation docs

### LINKED ISSUE #2298: [Docs][Navigation] Some documentation/details at the moment are not used in the published documentation
Some documentation/details at the moment are not used in the published documentation.
They will need to be reviewed and adjusted to be added to the TOC for the Navigation documentation.
https://github.com/unoplatform/uno.extensions/blob/4919dca542fe9de3a66d3526c9b50ff4643702af/doc/toc.yml#L73-L89

### PR DESCRIPTION
**Pull request by eriklimakc**
GitHub Issue (If applicable): closes #2298

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## What is the current behavior?

Unpublished documention regarding "Reference/Navigation"

## What is the new behavior?

Reviewed and updated documentation regarding "Reference/Navigation" to publish.

### CODE REVIEW

File: doc/Reference/Navigation/Design.md [hunk 1]
```diff
@@ -19,42 +19,22 @@ There are `INavigator` extension methods that accept a variety of parameters, de
...
+An application typically has one or more views responsible for controlling navigation. For example, a Frame that navigates between pages, or a TabBar that switches tabs
 
 Navigation controls can be categorized in three distinct groups with different Navigation goals.
 
 | Content-Based        | Has a content area that's used to display the current view                                                             |
 |----------------------|------------------------------------------------------------------------------------------------------------------------|
 | ContentControl       | Navigation creates an instance of a control and sets it as the Content                                                 |
 | Panel (eg Grid)      | Navigation sets a child element to Visible, hiding any previously visible child. Two scenarios:<br>	- An existing child is found. The child is set to Visible<br>	- No child is found. A new instance of a control is created and added to the Panel. The new instance is set to visible<br>Note that when items are hidden, they're no removed from the visual tree |
```
Comment (agneszitte) [inline doc/Reference/Navigation/Design.md:29]:
```suggestion
Navigation controls can be categorized into three distinct groups with different Navigation goals.

| Content-Based        | Has a content area that's used to display the current view                                                             |
|----------------------|------------------------------------------------------------------------------------------------------------------------|
| ContentControl       | Navigation creates an instance of a control and sets it as the Content                                                 |
| Panel (eg Grid)      | Navigation sets a child element to Visible, hiding any previously visible child. Two scenarios:<br>	- An existing child is found. The child is set to Visible<br>	- No child is found. A new instance of a control is created and added to the Panel. The new instance is set to visible<br>Note that when items are hidden, they're not removed from the visual tree |
```

File: doc/Reference/Navigation/Design.md [hunk 2]
```diff
@@ -19,42 +19,22 @@ There are `INavigator` extension methods that accept a variety of parameters, de
...
-| Frame                | Forward navigation adds a new page to the stack based <br>Backward navigation pops the current page off the stack<br>Combination eg forward navigation and clear back stack |
+| Frame                | Forward navigation adds a new page to the stack based <br>Backward navigation pops the current page off the stack<br>Combination, for example, forward navigation and clear back stack |
```
Comment (agneszitte) [inline doc/Reference/Navigation/Design.md:30]:
```suggestion
| Frame                | Forward navigation adds a new page to the stack-based <br>Backward navigation pops the current page off the stack<br>Combination, for example, forward navigation and clear back stack |
```

File: doc/Reference/Navigation/Qualifiers.md [hunk 3]
```diff
@@ -4,10 +4,15 @@ uid: Reference.Navigation.Qualifiers
...
-|-----------|--------------------------------------------------------------|------------------|--------------------------------------------------------------|
-| ""        | Navigate to page in frame, or open popup                     | "Home"           | Navigate to the HomePage                                     |
-| /         | Forward request to the root region                           | "/"<br>"/Login"  | Navigate to the default route at the root of navigation<br>Navigate to LoginPage at the root of navigation |
-| ./        | Forward request to child region                              | "./Info/Profile" | Navigates to the Profile view in the child region named Info |
-| !         | Open a dialog or flyout                                      | "!Cart"          | Shows the Cart flyout                                        |
-| -         | Back (Frame), Close (Dialog/Flyout) or respond to navigation | "-"<br>"--Profile"<br>"-/Login" | Navigate back one page (in a frame)<br>Navigate to Profile page and remove two pages from backstack<br>Navigate to Login page and clear backstack |
+When using navigation qualifiers can be used to make the navigating easier and smooth:
```
Comment (agneszitte) [inline doc/Reference/Navigation/Qualifiers.md:7]:
```suggestion
Navigation qualifiers can be utilized to make navigating easier and smoother:
```

File: doc/Reference/Navigation/Qualifiers.md [hunk 4]
```diff
@@ -4,10 +4,15 @@ uid: Reference.Navigation.Qualifiers
...
+|-----------|-------------------------------------------------|---------------------|--------------------------------------------------------------|
+| ""        | Navigate to page in frame or open popup        | "Home"              | Navigate to the HomePage                                     |
+| /         | Forward request to the root region              | "/"<br>"/Login"     | Navigate to the default route at the root of navigation<br>Navigate to LoginPage at the root of navigation |
+| ./        | Forward request to child region                 | "./Info/Profile"    | Navigate to the Profile view in the child region named Info |
+| !         | Open a dialog or flyout                         | "!Cart"             | Shows the Cart flyout                                        |
+| -         | Back (Frame), Close (Dialog/Flyout), or respond to navigation | "-"<br>"--Profile"<br>"-/Login" | Navigate back one page (in a frame)<br>Navigate to Profile page and remove two pages from backstack<br>Navigate to Login page and clear backstack |
```
Comment (agneszitte) [inline doc/Reference/Navigation/Qualifiers.md:15]:
```suggestion
| Qualifier | Description                                     | Example             | Example Usage                                                |
|-----------|-------------------------------------------------|---------------------|--------------------------------------------------------------|
| ""        | Navigate to page in frame or open popup        | "Home"              | Navigate to the `HomePage`                                     |
| /         | Forward request to the root region              | "/"<br>"/Login"     | Navigate to the default route at the root of navigation<br>Navigate to `LoginPage` at the root of navigation |
| ./        | Forward request to child region                 | "./Info/Profile"    | Navigate to the Profile view in the child region named Info |
| !         | Open a dialog or flyout                         | "!Cart"             | Shows the Cart flyout                                        |
| -         | Back (Frame), Close (Dialog/Flyout), or respond to navigation | "-"<br>"--Profile"<br>"-/Login" | Navigate back one page (in a frame)<br>Navigate to `ProfilePage` and remove two pages from back stack<br>Navigate to `LoginPage` and clear back stack |
```

File: doc/Reference/Navigation/Qualifiers.md [hunk 5]
```diff
@@ -4,10 +4,15 @@ uid: Reference.Navigation.Qualifiers
...
+
+> [!NOTE]
+> Besides using qualifiers as a string as part of the route specification, a `Qualifiers` class is also providade under the `Uno.Extensions.Navigation` namespace and can be specified when navigating, for example, `await navigator.NavigateViewModelAsync<MainViewModel>(this, Qualifiers.ClearBackStack);`.
```
Comment (agneszitte) [inline doc/Reference/Navigation/Qualifiers.md:18]:
```suggestion
> [!NOTE]
> Besides using qualifiers as a string as part of the route specification, a `Qualifiers` class is also provided under the `Uno.Extensions.Navigation` namespace and can be specified when navigating, for example, `await navigator.NavigateViewModelAsync<MainViewModel>(this, Qualifiers.ClearBackStack);`.
```

File: doc/Reference/Navigation/ViewMap.md [hunk 6]
```diff
@@ -1,6 +1,18 @@
...
 # What is a ViewMap
 
-- Define what a viewmap is and the relationship between view, viewmodel and datamap
+When registering routes, we can take advantage of the `ViewMap` object and its variations `DataViewMap` and `ResultDataViewMap`, to associate Views with ViewModels, specify the type of parameters ViewModels may take, and specify the type of return coming from a navigation.
+
+## ViewMap
+
+| Component      | Description          | Example |
+|----------------|----------------------|---------|
+| View           | The type of the View | `new ViewMap(typeof(MainPage))`<br><br>Alternatively can be specified as generic argument: `new ViewMap<MainPage>()` |
```
Comment (agneszitte) [inline doc/Reference/Navigation/ViewMap.md:13]:
```suggestion
| View           | The type of the View | `new ViewMap(typeof(MainPage))`<br><br>Alternatively can be specified as a generic argument: `new ViewMap<MainPage>()` |
```

File: doc/Reference/Navigation/NavigationRequestHandler.md
Comment (eriklimakc) [inline doc/Reference/Navigation/NavigationRequestHandler.md:1]: Will sync with Nick regarding content for this page.

File: doc/Reference/Navigation/ViewMap.md [hunk 7]
```diff
@@ -1,6 +1,18 @@
...
 
-- Define what a viewmap is and the relationship between view, viewmodel and datamap
+When registering routes, we can take advantage of the `ViewMap` object and its variations `DataViewMap` and `ResultDataViewMap`, to associate Views with ViewModels, specify the type of parameters ViewModels may take, and specify the type of return coming from a navigation.
+
+## ViewMap
+
+| Component      | Description          | Example |
+|----------------|----------------------|---------|
+| View           | The type of the View | `new ViewMap(typeof(MainPage))`<br><br>Alternatively can be specified as a generic argument: `new ViewMap<MainPage>()` |
+| ViewSelector   |         ?             | ? |
```
Comment (eriklimakc) [inline doc/Reference/Navigation/ViewMap.md:14]: Will sync with Nick regarding content for this part.

**Comment by eriklimakc** [comment]
Rebased to include https://github.com/unoplatform/uno.extensions/pull/2343

File: doc/Reference/Navigation/ViewMap.md (hunk 7 above)
Comment (nickrandolph) [reply doc/Reference/Navigation/ViewMap.md:14]: @eriklimakc we'll need to sync to finish this section off

File: doc/Reference/Navigation/ViewMap.md [hunk 8]
```diff
@@ -1,6 +1,18 @@
...
+| ViewModel      | The type of the ViewModel being associated with the View | `new ViewMap(typeof(MainPage), ViewModel: typeof(MainViewModel))`<br><br>Alternatively can be specified as a generic argument: `new ViewMap<MainPage, MainViewModel>()` |
+| Data           | A `DataMap<T>` to be injected into the ViewModel during navigation<br><br>Alternatively `DataViewMap` can be used | `new ViewMap<MainPage, MainViewModel>(Data: new DataMap<MyObject>)`<br><br>`new DataViewMap<MainPage, MainViewModel, MyObject>()` |
+| ResultData     | Define an association between the view and the type of data being requested<br><br>Alternatively `ResultDataViewMap` can be used | `new ViewMap<MainPage, MainViewModel>(ResultData: typeof(MyObject))`<br><br>`new ResultDataViewMap<MainPage, MainViewModel, MyObject>()` |
+| ViewAttributes |         ?             | ? |
```
[the remark this answers is not in the document]
Comment (eriklimakc) [reply doc/Reference/Navigation/ViewMap.md]: Work being done on https://github.com/unoplatform/uno.extensions/pull/2317. This page was removed.

**Comment by agneszitte** [comment]
@eriklimakc what is the status of this PR please

**Comment by eriklimakc** [comment]
> @eriklimakc what is the status of this PR please ? (based on latest)

@agneszitte I need to figure out how `NavigationRequestHandler` works to properly document it. https://github.com/unoplatform/uno.extensions/pull/2320#discussion_r1627667156

**Review by kazo0 — APPROVED** [review]

**Comment by kazo0** [comment]
@eriklimakc should this be moved to draft or can it merge as is?

File: doc/Reference/Navigation/Navigation.md [hunk 9]
```diff
@@ -1,6 +1,7 @@
 ---
 uid: Reference.Navigation.Overview
 ---
+
```
Comment (eriklimakc) [inline doc/Reference/Navigation/Navigation.md:4]: @kazo0 Do we actually need this page? It is used as a Overview page for the reference docs:

https://github.com/unoplatform/uno.extensions/blob/46b2d7fc4834cefd77adc5d5fd1a0663815be83c/doc/toc.yml#L76

Comment (kazo0) [reply doc/Reference/Navigation/Navigation.md:4]: We should have an Overview page that would open when clicking on the Navigation section in the TOC similar to what we do for MVUX in the Reference section: 

image

Comment (kazo0) [reply doc/Reference/Navigation/Navigation.md:4]: I would perhaps just use it as a landing page that could direct people to the individual pages in the Reference section for Navigation. Similar to this table: https://platform.uno/docs/articles/external/uno.extensions/doc/ExtensionsOverview.html

**Review by kazo0 — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

**Pull request merged**